File: sdk/python/tempmail_sdk/providers/thisisnotmyrealemail_com.py
"""
Mailinator 姊妹域名渠道：thisisnotmyrealemail.com
MX 指向 mail.mailinator.com，收信复用 mailinator 的 domain=public API
"""

from typing import List

from ..types import Email, EmailInfo
from . import mailinator

CHANNEL = "thisisnotmyrealemail-com"
DOMAIN = "thisisnotmyrealemail.com"


def generate_email() -> EmailInfo:
    """生成 thisisnotmyrealemail.com 域名的临时邮箱"""
    local = mailinator._random_string(12)
    return EmailInfo(channel=CHANNEL, email=f"{local}@{DOMAIN}")


def get_emails(token: str, email: str = "", **kwargs) -> List[Email]:
    """收信直接委托给 mailinator（domain=public 可读所有姊妹域名邮件）"""
    return mailinator.get_emails(token, email, **kwargs)

File: sdk/go/provider/thisisnotmyrealemail_com.go
package provider

// thisisnotmyrealemail.com：mailinator 官方姊妹域名，MX 指向 mail.mailinator.com。

// ThisisnotmyrealemailComGenerate 创建 thisisnotmyrealemail.com 临时邮箱
func ThisisnotmyrealemailComGenerate() (*CreatedMailbox, error) {
	local := randomStr(12)
	return &CreatedMailbox{
		Channel: "thisisnotmyrealemail-com",
		Email:   local + "@thisisnotmyrealemail.com",
	}, nil
}

// ThisisnotmyrealemailComGetEmails 读取邮件（复用 mailinator public API）
func ThisisnotmyrealemailComGetEmails(email string) ([]NormEmail, error) {
	return MailinatorGetEmails(email)
}
